qcengine/programs/nwchem/harvester.py

```diff
@@ -727,17 +751,40 @@ def harvest(in_mol: Molecule, nwout: str, **kwargs) -> Tuple[PreservingDict, Non
     """
 
     # Parse the NWChem output
-    outPsivar, outMol, outGrad, version, error = harvest_output(nwout)
+    out_psivar, out_mol, out_grad, version, error = harvest_output(nwout)
+
+    # If available, read higher-accuracy gradients
+    #  These were output using a Python Task in NWChem to read them out of the database
+    if outfiles.get("nwchem.grad") is not None:
+        logger.debug("Reading higher-accuracy gradients")
+        out_grad = json.loads(outfiles.get("nwchem.grad"))
+
+    # If available, read the hessian
+    out_hess = None
+    if outfiles.get("nwchem.hess") is not None:
+        out_hess = harvest_hessian(outfiles.get("nwchem.hess"))
 
     # Make sure the input and output molecules are the same
-    if outMol:
+    if out_mol:
         if in_mol:
-            if abs(outMol.nuclear_repulsion_energy() - in_mol.nuclear_repulsion_energy()) > 1.0e-3:
+            if abs(out_mol.nuclear_repulsion_energy() - in_mol.nuclear_repulsion_energy()) > 1.0e-3:
                 raise ValueError(
                     """NWChem outfile (NRE: %f) inconsistent with Psi4 input (NRE: %f)."""
-                    % (outMol.nuclear_repulsion_energy(), in_mol.nuclear_repulsion_energy())
+                    % (out_mol.nuclear_repulsion_energy(), in_mol.nuclear_repulsion_energy())
                 )
     else:
         raise ValueError("""No coordinate information extracted from NWChem output.""")
 
-    return outPsivar, None, outGrad, outMol, version, error
+    # If present, align the gradients and hessian with the original molecular coordinates
+    #  NWChem rotates the coordinates of the input molecule. `out_mol` contains the coordinates for the
+    #  rotated molecule, which we can use to determine how to rotate the gradients/hessian
+    amol, data = out_mol.align(in_mol, atoms_map=True, mols_align=True, verbose=0)
```

Comment:
Just to head off an insidious bug -- `atoms_map=True` means user water in HOH order is never transformed by NWChem into OHH. So long as that's true, it's good to keep this bool as True, but I wanted to check.

Reply:
That's a good point. I am not sure about whether NWC will ever reorder atoms, so I'm going to push a commit that sets `atom_map=False`

Reply:
Great. `False` will limit the size of mols that can be aligned in a timely manner, but grad/hess QC mols haven't been a problem yet.